src/Widgets/ScreenShield.vala

```diff
@@ -0,0 +1,433 @@
+//
+//  Copyright (C) 2020 elementary, Inc. (https://elementary.io)
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+namespace Gala {
+    [DBus (name = "org.freedesktop.login1.Manager")]
+    interface LoginManager : Object {
+        public signal void prepare_for_sleep (bool about_to_suspend);
+
+        public abstract GLib.ObjectPath get_session (string session_id) throws GLib.Error;
+
+        public abstract UnixInputStream inhibit (string what, string who, string why, string mode) throws GLib.Error;
+    }
+
+    [DBus (name = "org.freedesktop.login1.Session")]
+    interface LoginSessionManager : Object {
+        public abstract bool active { get; }
+
+        public signal void lock ();
+        public signal void unlock ();
+
+        public abstract void set_locked_hint (bool locked) throws GLib.Error;
+    }
+
+    public struct LoginDisplay {
+        string session;
+        GLib.ObjectPath objectpath;
+    }
+
+    [DBus (name = "org.freedesktop.login1.User")]
+    interface LoginUserManager : Object {
+        public abstract LoginDisplay? display { owned get; }
+    }
+
+    enum PresenceStatus {
+        AVAILABLE = 0,
+        INVISIBLE = 1,
+        BUSY = 2,
+        IDLE = 3
+    }
+
+    [DBus (name = "org.gnome.SessionManager.Presence")]
+    interface SessionPresence : Object {
+        public abstract uint status { get; }
+        public signal void status_changed (uint new_status);
+    }
+
+    [DBus (name = "org.freedesktop.DisplayManager.Seat")]
+    interface DisplayManagerSeat : Object {
+        public abstract void switch_to_greeter () throws GLib.Error;
+    }
+
+    public class ScreenShield : Clutter.Actor {
+        static ScreenShield? instance;
```

Comment:
```suggestion
        private ScreenShield? instance;
```
You don't really need it to be static, you'll always have just one ScreenShied per Gala process anyway, won't you?